### PR TITLE
Add a parameter for ignoring PDFs containing optional content blocks …

### PR DESCRIPTION
**Pull request by joelw**
Add a parameter for ignoring errors when parsing PDFs that contain optional content groups #28

### CODE REVIEW

File: lib/combine_pdf/parser.rb [hunk 1]
```diff
@@ -58,6 +59,7 @@ def initialize (string)
 			@strings_dictionary = {} # all strings are one string
 			@version = nil
 			@scanner = nil
+			@allow_optional_content = options[:allow_optional_content] || false
```
Comment (boazsegev) [inline lib/combine_pdf/parser.rb]: Is the `false` value necessary? the default missing key value is `nil`, which evaluates the same...

**Comment by boazsegev** [comment]
Looks cool 👍 

I commented on one thing, which is a nuance (just keeping the code beautiful)... but I think I'll merge the PR and update it later.

Thank you!

**Comment by boazsegev** [comment]
I updated the CHANGELOG to reflect the new feature, but I'm working on the decryption and form data retention (there's an issue with checkbox data being lost), so I'm waiting a bit with the new release.

If I can't fix these issues soon, I'll push a release anyway.

Thanks again :-)

**Comment by joelw** [comment]
I agree with your comment about the default nil - all good! Thank you very much - have a great weekend! :)

**Comment by kduraiswami** [comment]
Thank you for this

**Comment by Thomas-Malh** [comment]
Thanks for this option!

**Pull request merged**